recipes/grain/meta.yaml

```
@@ -44,7 +44,7 @@ requirements:
     - dm-tree
     - more-itertools >=9.1.0
     - numpy
-    - protobuf >=5.28.3
+    - protobuf >=5.28.3,<8.0.0
```

Can you add a comment pointing to the protobuf docs? Otherwise it is quite difficult to track where this `<8.0.0` comes from.

---

Not to be super pedantic, but usually these upper bounds are written like `<8.0dev0` or similar to avoid RCs and pre-releases getting in the way.

```suggestion
    - protobuf >=5.28.3,<8.0dev0
```

---

Thanks for the observation! @mtsokol probably it is better to do this before merging so the first build already have the correct metadata, thanks!

---

Done! I included `.rc0` that is the earliest RC which covers [all potential protobuf releases of 8.x](https://pypi.org/project/protobuf/#history).

---

> I can make it `8.0dev0`, I haven't found any dev releases for protobuf in its conda-forge feedstock or pypi.

That make sense, but please explain changes, as you may imagine reviewers can't read what you have in mind. : )

Anyhow, I guess the @jaimergp suggestion should cover also the `rc` case, so I think probably sticking to `8.0dev0` make sense.

---

Sorry about that! And thank you for all your assistance for this PR!